var/spack/repos/builtin/packages/trilinos/package.py

```diff
@@ -216,20 +221,6 @@ def url_for_version(self, version):
         url = "https://github.com/trilinos/Trilinos/archive/trilinos-release-{0}.tar.gz"
         return url.format(version.dashed)
 
-    # check that the combination of variants makes sense
-    def variants_check(self):
-        if ('+superlu-dist' in self.spec and
-            self.spec.satisfies('@11.14.1:11.14.3')):
-            # For Trilinos v11 we need to force SuperLUDist=OFF, since only the
-            # deprecated SuperLUDist v3.3 together with an Amesos patch is
-            # working.
```

Comment:
It would be good to keep some of these comments so we know why the `conflicts` are in place.